### ISSUE TITLE
Disallow primitives as types on root params/properties

### ISSUE THREAD

**Issue by brettz9** [gajus/eslint-plugin-jsdoc#505]
For params or properties like:

```js
/**
* @param {object} cfg
* @param {string} cfg.prop1
* @param {boolean} cfg.prop2
*/
```

...literals like `boolean`, `string`, `void`, etc. should not be allowed in the `object` position.

Could be added as a new rule `check-param-types`.
